en/micro-integrator/docs/develop/creating-artifacts/creating-a-proxy-service.md

```diff
@@ -58,6 +60,25 @@ Follow the instructions given below to create a new [Proxy Service](../../../ref
 
 The proxy service is created in the `src/main/synapse-config/proxy-services` folder under the project you specified.
 
+### Creating a Proxy Service using a WSDL definition
+
+Follow the steps given below after opening the **New Proxy Service** dialog box.
+
+1. Select **Generate Proxy Service using WSDL file** and click **Next**.
+
+    <img src="../../../assets/img/create_artifacts/new_proxy_service/create-new-proxy-option-wsdl.png" width="500">
+
+2. Provide a URL or a file location as the source of the WSDL and click **Finish**.
+
+    <img src="../../../assets/img/create_artifacts/new_proxy_service/create-proxy-from-wsdl.png" width="500">
+
+You will now see the mediation logic generated from the WSDL as shown below.
+
+<img src="../../../assets/img/create_artifacts/new_proxy_service/skeleton-proxy-service-wsdl.png" width="500">
+
+!!! Note
+    Switch cases are added only if the SOAPActions are defined in the provided WSDL definition.
```

Comment:
What do we mean by switch cases here? Is it specific to this mediation use case?
@arunans23

Reply:
WSDL has SOAPActions defined. Depending on the number of SOAPActions, we generate a Switch mediator skeleton with that many switch cases. The idea is when a user invokes the Proxy with a particular SOAPAction header, the mediation will happen through that particular switch case. (Eg: A request with SoapAction `Foo` comes in. Switch mediator will check the header and pass that message body to the relevant switch case block.)

This is of course optional. Users can delete all the skeleton code and rewrite everything on their own.